### PR TITLE
Compilation on Linux with Octave

### PR DESCRIPTION
**Pull request by gzsombor**

### CODE REVIEW

File: compile.m [hunk 1]
```diff
@@ -63,7 +63,23 @@ end
 
 if isunix
     disp('Unix');
-    % to come
+    include = ' -I/usr/include/opencv/ -I/usr/include/'; % /opt/local -> /usr/local
+    libpath = '/usr/lib/'; % /opt/local -> /usr/local
+
+    files = dir([libpath 'libcv.so']);
```
Comment (bilderbuchi) [inline compile.m:69]: Interestingly, here I had to say  
files = [dir([libpath 'libcv.so']) dir([libpath 'libcxcore.so'])];
to make it compile in matlab, ubuntu maverick 10.10 32bit. 
Just fyi, I don't yet know what to make of it.

File: mex/lk.cpp [hunk 2]
```diff
@@ -16,9 +16,9 @@
 // along with TLD.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "cv.h"
-#include "highgui.h"
```
Comment (bilderbuchi) [inline mex/lk.cpp:19]: why did you remove this?

Comment (gzsombor) [inline mex/lk.cpp:19]: Because there was no "highgui.h" in OpenCV 2.1, fortunately the code doesn't need it.

File: compile.m (hunk 1 above)
Comment (gzsombor) [inline compile.m:69]: hmm, yes, interesting. What is failing when that libcxcore.so is missing ? Unfortunately I'm not able run the code yet, so it is possible that you are right, and that cvcore is needed to

File: mex/lk.cpp (hunk 2 above)
Comment (bilderbuchi) [inline mex/lk.cpp:19]: you do know about the packages libhighgui2.1 and libhighgui-dev? (e.g. in ubuntu repo)
OTOH, good to not include it if it's not needed. :-)

File: compile.m (hunk 1 above)
Comment (bilderbuchi) [inline compile.m:69]: can't tell you right now, only got matlab at work. it was complaining about an undeclared function. a quick googling showed that said function was included in libcxcore.so, so including it solved it.

**Comment by xamox** [comment]
I tried to run your version on Ubuntu 10.10-64bit but got the following error when running: octave run_TLD.m

parse error near line 27 of file /home/xamox/Code/OpenTLD/tld/tldExample.m

  invalid parameter list

> > > finish = 0; function handleKey(~,~), finish = 1; end % by pressing any key, the process will exit

I ran 'octave compile.m' before I ran the example.
Any ideas what maybe causing this?

**Comment by bilderbuchi** [comment]
I get the same error (and I'd think gzsombor, too). Fixing errors like this one is the next step, up until now the aim (i think) was to get it to compile properly. Not yet an idea what causes this.

**Comment by bilderbuchi** [comment]
matlab says
??? Error: File: tldExample.m Line: 27 Column: 32
Unexpected MATLAB operator.
So this is not even correct syntax in Matlab (R2009a). I'm not sure what is going on here.

**Comment by bilderbuchi** [comment]
Fixed this first error by defining dummy arguments. Fixed by https://github.com/bilderbuchi/OpenTLD/commit/1caa54c7eb5b85a86344dafdb62c5e2573efa854

Turns out, this is valid syntax from R2009b and beyond: http://blogs.mathworks.com/steve/2010/01/11/about-the-unused-argument-syntax-in-r2009b/ and octave >=3.4

File: compile.m (hunk 1 above)
Comment (bilderbuchi) [inline compile.m:69]: probably from a wrong opencv version. turns out current ubuntu package is only 2.1, not yet 2.2.

**Comment by zk00006** [comment]
Is this branch ready to be pulled back? Have you made modifications that will work with Matlab as well or is it more specific for Octave only ?

**Comment by zk00006** [comment]
Is this branch ready to be pulled back? Have you made modifications that will work with Matlab as well or is it more specific for Octave only ?

**Comment by zk00006** [comment]
Is this branch ready to be pulled back? Have you made modifications that will work with Matlab as well or is it more specific for Octave only ?

**Comment by bilderbuchi** [comment]
I'm not sure, probably, but gzsombor should comment on this. 
With my further work based on the state of this pull request, I tried (and will continue to do so) to preserve matlab functionality by using octave/matlab-dependent flags (https://github.com/bilderbuchi/OpenTLD/commit/b0d89e5b1d8a4b6f5f0326ace056c878b7934e16). Also, I'll try to include as much multi-platform work as possible, though I can primarily test/code on Ubuntu, and not at all on MacOS. My branch is not ready for a pull, though, since I discovered that much of what I did may be in vain (errors because of old versions). Will compile new opencv and octave and check again.

Preserving one main branch with multi-platform and multi-client capabilities is the only way this makes sense, in my opinion. Not least because we keep the ability to test/compare results against each other.
